test/unit/model/helpers/downloader-test.js

```diff
@@ -271,4 +258,40 @@ describe('Downloader', function() {
       expect(successHandler).to.be.calledOnce;
     });
   });
+
+  describe('restartDownload', function(){
+    let options = 'http://example.com/jdk.zip';
+
+    it('should change downloader status from \'Download Failed\' to \'Downloading\'', function(){
+      let spy = sandbox.spy(fakeProgress, 'setStatus');
+      downloader.restartDownload();
+      expect(spy).to.have.been.calledOnce;
+      expect(downloader.downloadSize).to.be.equal(0);
+      expect(downloader.received ).to.be.equal(0);
+      expect(downloader.currentSize  ).to.be.equal(0);
+      expect(fakeProgress.setStatus).to.have.been.calledOnce;
+      expect(fakeProgress.setStatus).to.have.been.calledWith('Downloading');
+    });
+
+    it('should call authDownload for entries that requires authentication', function(){
+      let requestGetSpy = sandbox.spy(request, 'get');
+      downloader.setWriteStream(new PassThrough());
+      downloader.restartDownload();
+      downloader.downloadAuth(options, 'username', 'password', 'key', 'sha');
+      expect(requestGetSpy).to.be.calledOnce;
+      expect(requestGetSpy.args[0][0].url).to.be.equal(options, 'username', 'password', 'key', 'sha');
+
+    });
+
+    it('should call download method for entries that does not require authentication',function(){
```

Comment:
Basically it should be the same as test for `downloadAuth`, but use `download` method instead.